spyder/plugins/ipythonconsole/utils/kernelspec.py

```diff
@@ -125,6 +126,17 @@ def argv(self):
             # If executable is a conda environment, use "run" subcommand to
             # activate it and run spyder-kernels.
             conda_exe = find_conda()
+            if not conda_exe:
+                # Raise error since we were unable to determine the path to
+                # the conda executable (e.g conda was installed in a
+                # non-standard location).
+                # See spyder-ide/spyder#23595
+                raise SpyderKernelError(
+                    _("Unable to determine path to conda executable. "
+                      "Please add the path to the conda executable to your "
+                      "PATH environment variable for it to be detected.")
```

Comment:
```suggestion
                    _(
                        "Spyder couldn't find conda or mamba in your system "
                        "to activate the kernel's environment. Please add the "
                        "directory where the conda or mamba executable is "
                        "located to your PATH environment variable for it to "
                        "be detected."
                    )
```

Improve message a bit.